File: API/RequestHelpers/ProductParams.cs
using System;

namespace API.RequestHelpers;

public class ProductParams : PaginationParams
{
    public string? OrderBy { get; set; }
    public string? SearchTerm { get; set; }
    public string? Generos { get; set; }
    public string? Anos { get; set; }
    public string? Marcas { get; set; }
    public string? Modelos { get; set; }
    public string? Tipos { get; set; }
    public string? Capacidades { get; set; }
    public string? Cores { get; set; }
    public string? Materiais { get; set; }
    public string? Tamanhos { get; set; }
    public string? CategoryIds { get; set; }
    public string? CampaignIds { get; set; }
    public bool? HasDiscount { get; set; }
}

File: client/src/app/models/productParams.ts
export type ProductParams = {
    orderBy: string;
    searchTerm?: string;
    generos?: string[];
    anos?: number[];
    categoryIds?: number[];
    campaignIds?: number[];
    marcas?: string[];
    modelos?: string[];
    tipos?: string[];
    capacidades?: string[];
    cores?: string[];
    materiais?: string[];
    tamanhos?: string[];
    hasDiscount?: boolean;
    pageNumber: number;
    pageSize: number;
}